kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -26,59 +27,59 @@ const ContentNodeResource = coreApp.resources.ContentNodeResource;
 const FacilityUserResource = coreApp.resources.FacilityUserResource;
 const SummaryLogResource = coreApp.resources.ContentSummaryLogResource;
 
+/**
+ * Helper function for _showChannelList
+ * @param {object} channel - to get recentActivity for
+ * @param {string} classId -
+ * @returns {Promise} that resolves channel with lastActive value in object:
+ *   { 'channelId': dateOfLastActivity }
+*/
+function channelLastActivePromise(channel, classId) {
+  const summaryPayload = {
+    channel_id: channel.id,
+    collection_kind: ReportConstants.UserScopes.CLASSROOM,
+    collection_id: classId,
+  };
 
-function _showChannelList(store, classId) {
+  // workaround for conditionalPromise.then() misbehaving
+  return new Promise(
+    (resolve, reject) => {
+      const getSumm = ContentSummaryResource.getModel(channel.root_id, summaryPayload).fetch();
+      getSumm.then(
+        channelSummary => {
+          resolve({ [channel.id]: channelSummary.last_active });
+        },
+        error => reject(error)
+      );
+    }
+  );
+}
+
+function getAllChannelsLastActivePromise(channels, classId) {
+  const promises = channels.map((channel) => channelLastActivePromise(channel, classId));
+  return Promise.all(promises);
+}
+
+function _showChannelList(store, classId, showRecentOnly = false) {
```

Comment:
I think that would be useful if we ever want to toggle the recency filter. I agree the code is kind of awkward adding this new argument ad-hoc, but it's the quickest way to extend the code to fit this feature.